### PR TITLE
AST-653 - Added compatibility for search block in WordPress 5.8

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Added compatibility for search block introducing in WP-5.8
- As this block is in WP-5.7 versions as well, added backward for old users to not affect anything directly

### Screenshots
- with master version - https://share.bsf.io/nOuP7OrG
- with this branch - https://share.bsf.io/v1uY9kkx

### Types of changes
- Added GB editor + Frontend support for search block

### How has this been tested?
- Check with different search box styles (search block)
- Check on all devices
- Check GB editor view
- Button styles should work for search buttons
- Check search block button with text/icons
- Check different search button styles provided in the block

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by shubhw12 — APPROVED** [review]

File: inc/dynamic-css/wordpress-compat.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * @package astra
+ * @since x.x.x
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+add_filter( 'astra_dynamic_theme_css', 'astra_wordpress_compat_css' );
```
Comment (premanshup) [inline inc/dynamic-css/wordpress-compat.php:13]:
```suggestion
add_filter( 'astra_dynamic_theme_css', 'astra_search_block_css' );
```

File: inc/dynamic-css/wordpress-compat.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<?php
```
Comment (premanshup) [inline inc/dynamic-css/wordpress-compat.php:1]: Can we change the file name to search-block-css.php

File: inc/dynamic-css/wordpress-compat.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		.site-content form.wp-block-search .wp-block-search__inside-wrapper .wp-block-search__input {
+			padding: 12px;
+		}
+		.wp-block-search__button svg {
+            fill: currentColor;
+			width: 20px;
+			height: 20px;
+        }';
+
+		return $dynamic_css .= Astra_Enqueue_Scripts::trim_css( $desktop_css );
```
Comment (premanshup) [inline inc/dynamic-css/wordpress-compat.php:45]: @imnavanath Are we applying CSS to tablet and mobile same desktop CSS?

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/dynamic-css/wordpress-compat.php (hunk 3 above)
Comment (imnavanath) [reply inc/dynamic-css/wordpress-compat.php]: Not we get any case where responsive CSS is needed, that's why I have renamed this var now with `$compatibility_css`

File: inc/theme-update/class-astra-theme-background-updater.php [hunk 4]
```diff
@@ -95,6 +95,7 @@ class Astra_Theme_Background_Updater {
 			),
 			'3.7.0' => array(
 				'astra_fix_footer_widget_right_margin_case',
+				'astra_support_wp_5_8',
```
[the remark this answers is not in the document]
Comment (imnavanath) [reply inc/theme-update/class-astra-theme-background-updater.php:105]: Support is for both search block & WordPress 5.8 so likely updated filter & file names.

File: inc/theme-update/astra-update-functions.php [hunk 5]
```diff
@@ -3100,3 +3100,19 @@ function astra_fix_footer_widget_right_margin_case() {
...
+ *
+ * @since x.x.x
+ * @return void
+ */
+function astra_support_wp_5_8() {
+	$theme_options = get_option( 'astra-settings' );
+
+	// Set flag on existing user's site to not reflect changes directly.
+	if ( ! isset( $theme_options['astra-suport-wp-5-8'] ) ) {
+		$theme_options['astra-suport-wp-5-8'] = false;
```
[the remark this answers is not in the document]
Comment (imnavanath) [reply inc/theme-update/astra-update-functions.php:3145]: updated flag name to `'astra-support-block-editor'` 👍

File: inc/class-astra-dynamic-css.php [hunk 6]
```diff
@@ -3163,6 +3172,18 @@ public static function elementor_default_color_font_setting() {
 
 		}
 
+		/**
+		 * For existing users, do not reflect direct change.
+		 *
+		 * @since x.x.x
+		 * @return boolean true if WordPress-5.8 compatibility enabled, False if not.
+		 */
+		public static function is_wordpress_5_8_support_enabled() {
```
[the remark this answers is not in the document]
Comment (imnavanath) [reply inc/class-astra-dynamic-css.php:3229]: updated function name `is_block_editor_support_enabled`

File: inc/class-astra-dynamic-css.php [hunk 7]
```diff
@@ -3163,6 +3172,18 @@ public static function elementor_default_color_font_setting() {
...
+			$astra_settings                        = get_option( ASTRA_THEME_SETTINGS );
+			$astra_settings['astra-suport-wp-5-8'] = ( isset( $astra_settings['astra-suport-wp-5-8'] ) && false === $astra_settings['astra-suport-wp-5-8'] ) ? false : true;
+			return apply_filters( 'astra_wordpress_5_8_support', $astra_settings['astra-suport-wp-5-8'] );
```
[the remark this answers is not in the document]
Comment (imnavanath) [reply inc/class-astra-dynamic-css.php]: Done

File: inc/dynamic-css/wordpress-compat.php (hunk 1 above)
Comment (imnavanath) [reply inc/dynamic-css/wordpress-compat.php]: done

File: inc/dynamic-css/wordpress-compat.php (hunk 2 above)
Comment (imnavanath) [reply inc/dynamic-css/wordpress-compat.php]: done

**Review by premanshup — APPROVED** [review]

**Pull request merged**